src/dmd/semantic2.d

```diff
@@ -349,6 +350,88 @@ private extern(C++) final class Semantic2Visitor : Visitor
 
         fd.semanticRun = PASSsemantic2;
 
+        //printf("FuncDeclaration::semantic2 [%s] fd0 = %s %s\n", this.loc.toChars(), this.toChars(), this.type.toChars());
+        if (fd.overnext && !fd.errors)
+        {
+            OutBuffer buf1;
+            OutBuffer buf2;
+
+            // Always starts the lookup from 'this', because the conflicts with
+            // previous overloads are already reported.
+            auto f1 = fd;
+            mangleToFuncSignature(&buf1, f1);
+
+            overloadApply(f1, (Dsymbol s)
+            {
+                auto f2 = s.isFuncDeclaration();
+                if (!f2 || f1 == f2 || f2.errors)
+                    return 0;
+
+                // Don't have to check conflict between declaration and definition.
+                if ((f1.fbody !is null) != (f2.fbody !is null))
+                    return 0;
+
+                /* Check for overload merging with base class member functions.
+                 *
+                 *  class B { void foo() {} }
+                 *  class D : B {
+                 *    override void foo() {}    // B.foo appears as f2
+                 *    alias foo = B.foo;
+                 *  }
+                 */
+                if (f1.overrides(f2))
+                    return 0;
+
+                // extern (C) functions always conflict each other.
+                if (f1.ident == f2.ident &&
+                    f1.toParent2() == f2.toParent2() &&
+                    (f1.linkage != LINKd && f1.linkage != LINKcpp) &&
+                    (f2.linkage != LINKd && f2.linkage != LINKcpp))
+                {
+                    /* Allow the hack that is actually used in druntime,
+                     * to ignore function attributes for extern (C) functions.
+                     * TODO: Must be reconsidered in the future.
```

Comment:
File an issue for this, and add link to the issue?

Reply:
-> https://issues.dlang.org/show_bug.cgi?id=18206